### PR TITLE
Fix API Inconsistencies, Closes #18

### LINKED ISSUE #18: Fix API Inconsistencies
@sergiorussia I have a favor to ask.  I want you to break our API. 😄 

As SansORM has evolved over time, some minor inconsistencies have crept in; due to expendiency as well as lack of clarity about the separation of concerns (I'm guilty on both counts).

Because *master* already contains breaking changes (the removal of deprecated methods), the next release needs to go out as v3.x.x.  So, now is the perfect time to make unnecessary work for myself ... er ... the perfect time to move things around.

### Background
As originally conceived, ``OrmElf`` was to contain methods that deal in the currency of "objects" (/classes), but know nothing of where ``Connection``, ``Statement``, or ``ResultSet`` objects come from.  For example:
```java
public static <T> T resultSetToObject(ResultSet resultSet, T target) throws SQLException

public static <T> List<T> listFromClause(Connection connection, Class<T> clazz, String clause, Object... args) throws SQLException
...
```

Whereas ``SqlClosureElf`` knows how to acquire connections (using ``SqlClosure``), and in addition to "object" convenience methods, might also contain *pure SQL* convenience methods.

### Inconsistencies
Unfortunately, some *pure SQL* functionality has crept into ``OrmElf``, and I would like to see it moved to ``SqlClosureElf``.  The ones I have identified are these:
```java
public static String getInClausePlaceholders(final String...items)

public static ResultSet executeQuery(Connection connection, String sql, Object... args) throws SQLException

public static int executeUpdate(Connection connection, String sql, Object... args) throws SQLException

public static Number numberFromSql(Connection connection, String sql, Object... args) throws SQLException
```

Do you feel like refactoring those?

### PR DESCRIPTION
**Pull request by sergiorussia**
Closes #18.

- dropped deprecated methods
- moved sql-only related methods from OrmElf to SqlClosureElf
- added SansOrm configuration facade
- used assertj in tests

### CODE REVIEW

**Comment by sergiorussia** [comment]
@brettwooldridge this PR covers most parts of the issue. as soon as i'll remember other things i wanted to add - i'll add them

**Comment by brettwooldridge** [comment]
@sergiorussia Nice work!

Two minor things so far.

Given that ``SansOrm`` is now used to initialize the ``DataSource`` and ``TransactionManager`` in the ``SqlClosure`` class, I suggest making that the *only* way to do so, by making ``SqlClosure.setDefaultDataSource()`` package-level access.  But I think it's OK to leave the public ``SqlClosure(final DataSource ds)`` constructor though.

Next, can you fix these JavaDoc warnings?
```
[WARNING] Javadoc Warnings
[WARNING] /Users/brettw/Documents/dev/SansOrm/src/main/java/com/zaxxer/sansorm/SansOrm.java:16: warning: no @param for dataSource
[WARNING] public static DataSource initializeTxNone(DataSource dataSource) {
[WARNING] ^
[WARNING] /Users/brettw/Documents/dev/SansOrm/src/main/java/com/zaxxer/sansorm/SansOrm.java:16: warning: no @return
[WARNING] public static DataSource initializeTxNone(DataSource dataSource) {
[WARNING] ^
[WARNING] /Users/brettw/Documents/dev/SansOrm/src/main/java/com/zaxxer/sansorm/SansOrm.java:22: warning: no @param for dataSource
[WARNING] public static DataSource initializeTxSimple(DataSource dataSource) {
[WARNING] ^
[WARNING] /Users/brettw/Documents/dev/SansOrm/src/main/java/com/zaxxer/sansorm/SansOrm.java:22: warning: no @return
[WARNING] public static DataSource initializeTxSimple(DataSource dataSource) {
[WARNING] ^
[WARNING] /Users/brettw/Documents/dev/SansOrm/src/main/java/com/zaxxer/sansorm/SansOrm.java:28: warning: no @param for dataSource
[WARNING] public static DataSource initializeTxCustom(DataSource dataSource, TransactionManager txManager, UserTransaction userTx) {
[WARNING] ^
[WARNING] /Users/brettw/Documents/dev/SansOrm/src/main/java/com/zaxxer/sansorm/SansOrm.java:28: warning: no @param for txManager
[WARNING] public static DataSource initializeTxCustom(DataSource dataSource, TransactionManager txManager, UserTransaction userTx) {
[WARNING] ^
[WARNING] /Users/brettw/Documents/dev/SansOrm/src/main/java/com/zaxxer/sansorm/SansOrm.java:28: warning: no @param for userTx
[WARNING] public static DataSource initializeTxCustom(DataSource dataSource, TransactionManager txManager, UserTransaction userTx) {
[WARNING] ^
[WARNING] /Users/brettw/Documents/dev/SansOrm/src/main/java/com/zaxxer/sansorm/SansOrm.java:28: warning: no @return
[WARNING] public static DataSource initializeTxCustom(DataSource dataSource, TransactionManager txManager, UserTransaction userTx) {
[WARNING] ^
```

File: src/main/java/com/zaxxer/sansorm/SqlClosure.java [hunk 1]
```diff
@@ -360,36 +330,6 @@ public static void quietClose(ResultSet resultSet)
...
-    * Used to automatically code a ResultSet when the closure completes.
-    *
-    * @param resultSet the ResultSet to automatically close
-    * @return the ResultSet that will be closed (same as the input parameter)
-    */
-   @Deprecated
-   protected final ResultSet autoClose(ResultSet resultSet)
-   {
-      if (resultSet != null) {
-         closeResultSets.add(resultSet);
```
Comment (brettwooldridge) [inline src/main/java/com/zaxxer/sansorm/SqlClosure.java]: With the removal of ``autoClose(Statement)/autoClose(ResultSet)`` you can also remove the ``closeStatements`` and ``closeResultSets`` class members.

Comment (sergiorussia) [reply src/main/java/com/zaxxer/sansorm/SqlClosure.java]: @brettwooldridge, fixed all mentioned issues

**Comment by brettwooldridge** [comment]
@sergiorussia If you don't have any more changes, I'm ready to squash...

**Comment by sergiorussia** [comment]
@brettwooldridge go for it, no more changes at the moment

**Comment by brettwooldridge** [comment]
@sergiorussia BTW, how is SansOrm working out for you in your project?

**Comment by sergiorussia** [comment]
@brettwooldridge works just as it should - smoothly and fast :) since we migrated from Hibernate, all datasource-related issues have gone away.

**Comment by brettwooldridge** [comment]
@sergiorussia Awesome, I love hearing that!

**Pull request merged**